src/providers/virtual/qgsvirtuallayerprovider.cpp

```diff
@@ -132,6 +132,7 @@ bool QgsVirtualLayerProvider::loadSourceLayers()
       connect( vl, &QgsVectorLayer::featureAdded, this, &QgsVirtualLayerProvider::invalidateStatistics );
       connect( vl, &QgsVectorLayer::featureDeleted, this, &QgsVirtualLayerProvider::invalidateStatistics );
       connect( vl, &QgsVectorLayer::geometryChanged, this, &QgsVirtualLayerProvider::invalidateStatistics );
+      connect( vl, &QgsVectorLayer::updatedFields, this, [ = ] { createVirtualTable( qobject_cast<QgsVectorLayer *>( sender() ) ); } );
```

Comment:
Instead of relying on sender() here, you could capture vl in the lambda and use it directly